src/core/webdavconnection.cpp

```diff
@@ -652,70 +652,73 @@ void WebdavConnection::downloadPath( const QString &localPath )
   }
 }
 
-void WebdavConnection::uploadPath( const QString &localPath )
+void WebdavConnection::uploadPaths( const QStringList &localPaths )
 {
-  QFileInfo fi( QDir::cleanPath( localPath ) );
-  QDir dir( fi.isFile() ? fi.absolutePath() : fi.absoluteFilePath() );
-
-  bool webdavConfigurationExists = dir.exists( "qfield_webdav_configuration.json" );
-  QStringList remoteChildrenPath;
-  while ( !webdavConfigurationExists )
+  for ( const QString &localPath : localPaths )
```

Comment:
Please remove parts that shouldn't be repeated again and again as you loop through the local paths.

The whole discovery of the qfield_webdav_configuration.json file should be done only one time. Similarly setting the mProcessRemotePath and mProcessLocalPath should be done only once. You'll probably still need to loop through all the files to find the item path that is parent-most  (i.e. take the following list:

- /project/attachments
- /project/my.txt

The mProcessRemotePath and mProcessLocalPath should be /project , not project/attachments

Also, thinking out-loud here, we'll need to detect and bail out when someone provides items from different webdav configuration, that wouldn't work at all ATM.